qiskit_nature/drivers/molecule.py

```diff
@@ -76,6 +79,72 @@ def __init__(
 
         self._perturbations = None  # type: Optional[List[float]]
 
+    def to_hdf5(self, parent: h5py.Group) -> None:
+        """Stores this instance in an HDF5 group inside of the provided parent group.
+
+        See also :func:`~qiskit_nature.hdf5.HDF5Storable.to_hdf5` for more details.
+
+        Args:
+            parent: the parent HDF5 group.
+        """
+        group = parent.require_group(self.__class__.__name__)
+        group.attrs["__class__"] = self.__class__.__name__
+        group.attrs["__module__"] = self.__class__.__module__
+        group.attrs["__version__"] = self.VERSION
+
+        geometry_group = group.create_group("geometry", track_order=True)
+        for idx, geom in enumerate(self._geometry):
+            symbol, coords = geom
+            geometry_group.create_dataset(str(idx), data=coords)
+            geometry_group[str(idx)].attrs["symbol"] = symbol
+
+        group.attrs["units"] = self.units.value
+        group.attrs["multiplicity"] = self.multiplicity
+        group.attrs["charge"] = self.charge
+
+        if self._masses:
+            group.create_dataset("masses", data=self._masses)
+
+    @staticmethod
+    def from_hdf5(h5py_group: h5py.Group) -> Molecule:
+        """Constructs a new instance from the data stored in the provided HDF5 group.
+
+        See also :func:`~qiskit_nature.hdf5.HDF5Storable.from_hdf5` for more details.
+
+        Args:
+            h5py_group: the HDF5 group from which to load the data.
+
+        Returns:
+            A new instance of this class.
+        """
+        geometry = []
+        for atom in h5py_group["geometry"].values():
+            geometry.append((atom.attrs["symbol"], list(atom[...])))
+
+        units: UnitsType  # pylint: disable=unused-variable
+        for unit in UnitsType:
+            if unit.value == h5py_group.attrs["units"]:
+                units = unit
+                break
+        else:
+            units = UnitsType.ANGSTROM
+
+        # TODO: leverage units after https://github.com/Qiskit/qiskit-nature/issues/477
```

Comment:
The referenced issue in this TODO has already been closed. Should we take care of units here now ahead of merging lest it gets overlooked.